src/main/java/hu/bme/mit/spaceship/TorpedoStore.java

```diff
@@ -36,8 +37,7 @@ public boolean fire(int numberOfTorpedos){
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
-    double r = generator.nextDouble();
+    double r = this.rand.nextInt();
```

Comment:
Farkas hiba, ezért bukik a teszt! A rand.nextDouble() 0.0-1.0-ig ad értéket, ami mindig nagyobb vagy egyenlő 0.0, de a nextInt() negatív is lehet.